gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/auth/provider/ClientBasicAuthProvider.java

```diff
@@ -72,10 +81,32 @@ public void handle(Client client, RoutingContext context, Handler<AsyncResult<Cl
             }
             String clientId = urlDecode(decoded.substring(0, colonIdx));
             String clientSecret = urlDecode(decoded.substring(colonIdx + 1));
-            if (!client.getClientId().equals(clientId) || !client.getClientSecret().equals(clientSecret)) {
+
+            if (!client.getClientId().equals(clientId)) {
                 handler.handle(Future.failedFuture(new InvalidClientException(ClientAuthHandler.GENERIC_ERROR_MESSAGE, authenticationHeader())));
                 return;
             }
+
+            if (!isEmpty(client.getClientSecrets())) {
+                // take the first one as for now, we do not manage multiple secrets
+                ClientSecret hashedSecret = client.getClientSecrets().get(0);
+                var pwdEncoder = client.getSecretSettings()
+                        .stream().filter(settings -> settings.getId().equals(hashedSecret.getSettingsId()))
+                        .findFirst()
+                        .map(appSecretService::getOrCreatePasswordEncoder)
+                        .orElseGet(() -> appSecretService.getOrCreateNoOpPasswordEncoder());
```

Comment:
to remove codesmell: 
.orElseGet(appSecretService::getOrCreateNoOpPasswordEncoder);